misc/AL2023-Dockerfile

```diff
@@ -19,6 +19,12 @@ RUN dnf install -y \
     automake \
     autoconf
 
+# Some basic tools that are typical during builds
```

Comment:
Can we fold this list into the previous one, on line 13? There's no reason to have two lists, the first one already covers "typical package build deps".